Example for requesting access token (#26573)
Azure/azure-sdk-for-net: sdk/identity/Azure.Identity/README.md

var eventHubProducerClient = new EventHubProducerClient("myeventhub.eventhubs.windows.net", "myhubpath", credential);
```

> ### Requesting access token for Azure resource with scope

Reviewer: ```suggestion
### Request access token for Azure resource with scope

```

Sometimes an access token is needed as a bearer token in an OAuth 2.0 flow. The `DefaultAzureCredential` can be used to retrive the access token via the `TokenRequestContext`.

```C# Snippet:AccessToken
var credential = new DefaultAzureCredential();
var token = await credential.GetTokenAsync(
  new TokenRequestContext(new[] { "https://myvault.vault.azure.net/.default" }));

> ```

Reviewer: This snippet needs to be defined in the code somewhere. That allows us to ensure the code snippet compiles. @christothes Where should this snippet be defined?

Reviewer 2: Thanks for catching this - They should go [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/tests/samples/ReadmeSnippets.cs)

Note that once you add it there, you should run this from the root dir of your repo clone:
`.\eng\scripts\Update-Snippets.ps1 identity`

Author: @christothes is there any specific way I should name the code snippet, so it is linked? Or should I just come up with a meaningful name of the method?

Reviewer 2: The names need to match in the .md and the .cs file, but the content of the snippet will be auto-populated by the script when it is run (so that it is always gauranteed to match the code).

The name should just be descriptive, but the specific name is not important. I'd suggest something like `CallingGetTokenDirectly`.

> > The scope `.default` is required in [client credentials](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow) and [On-Behalf-Of](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow) flows as there are not user interaction.

Reviewer: ```suggestion
> The scope `.default` is required in [client credentials](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow) and [On-Behalf-Of](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow) flows, as there isn't user interaction.
```
